File: Backend/Models/Ticket.cs
namespace ProjetV1.Models
{
    public class Ticket
    {
        public int Id { get; set; }
        public int UtilisateurId { get; set; }
        public Utilisateur Utilisateur { get; set; }

        public int MatchId { get; set; }
        public Match Match { get; set; }
        public double Prix {  get; set; }
        public string Statut { get; set; } // Dispo ou Vendu
        public int NumPlace { get; set; }
        public Paiement Paiement { get; set; }

    }
}

File: backend/app/Models/Ticket.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ticket extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'match_id',
        'prix',
        'statut',
        'numero_place',
    ];

    // Relation to User
    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    // Relation to Match
    public function match()
    {
        return $this->belongsTo(\App\Models\Matches::class, 'match_id');
    }
}
